src/ucp/tag/eager_snd.c

```diff
@@ -266,19 +266,25 @@ static ucs_status_t ucp_tag_eager_sync_zcopy_multi(uct_pending_req_t *self)
     ucp_eager_sync_first_hdr_t first_hdr;
     ucp_eager_middle_hdr_t middle_hdr;
 
-    first_hdr.super.super.super.tag = req->send.msg_proto.tag.tag;
+    if (req->send.state.dt.offset != 0) {
+        middle_hdr.msg_id = req->send.msg_proto.message_id;
+        middle_hdr.offset = req->send.state.dt.offset;
+
+        return ucp_do_am_zcopy_multi(self, UCP_AM_ID_LAST,
+                                     UCP_AM_ID_EAGER_MIDDLE, NULL, 0,
+                                     &middle_hdr, sizeof(middle_hdr), NULL, 0ul,
+                                     ucp_tag_eager_sync_zcopy_req_complete, 1);
+    }
```

Comment:
why need to change

Reply:
to not get request ID for each segment, because after sending the first fragment, ID will be released